### ISSUE TITLE
Import maps don't use SystemJS fetch hook

### ISSUE THREAD

**Issue by erikt9** [systemjs/systemjs#2374]
### Import maps should use fetch hook to retrieve script source

The relevant code from `import-maps.js`:

```
if (script.type === 'systemjs-importmap') {
      script.sp = true;
      var fetchPromise = script.src ? fetch(script.src, { integrity: script.integrity }).then(function (res) {
        if (!res.ok)
          throw Error(process.env.SYSTEM_PRODUCTION ? res.status : 'Invalid status code: ' + res.status);
        return res.text();
```

Seems like the `fetch` here should be using the fetch hook instead of the global method?  Would be nice for consistency if all fetches SystemJS performs go through the hook.

**Comment by joeldenning** [comment]
Agreed, I'll create a pull request to implement this.

**Comment by joeldenning** [comment]
I've implemented this in #2376

**Comment by joeldenning** [comment]
Published in https://github.com/systemjs/systemjs/releases/tag/6.12.1
